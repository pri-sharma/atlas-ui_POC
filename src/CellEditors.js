import currency from 'currency.js'

// gets called once before the renderer is used
NumericCellEditor.prototype.init = function (params) {
    // create the cell
    this.eInput = document.createElement('input');
    params.eGridCell.style.height = '25px'; // make cell stay the same size

    if (isCharNumeric(params.charPress)) {
        this.eInput.value = params.charPress;
    } else {
        if (params.value !== undefined && params.value !== null) {
            this.eInput.value = params.value;
        }
    }

    let that = this;
    this.eInput.addEventListener('keypress', function (event) {
        if (!isKeyPressedNumeric(event)) {
            that.eInput.focus();
            if (event.preventDefault) event.preventDefault();
        }
        else if (that.isKeyPressedNavigation(event)){
            event.stopPropagation();
        }
    });

    // only start edit if key pressed is a number, not a letter
    this.cancelBeforeStart = params.charPress && ('1234567890'.indexOf(params.charPress) < 0);
};

NumericCellEditor.prototype.isKeyPressedNavigation = function (event){
    return event.keyCode===39
        || event.keyCode===37;
};


// gets called once when grid ready to insert the element
NumericCellEditor.prototype.getGui = function () {
    return this.eInput;
};

// focus and select can be done after the gui is attached
NumericCellEditor.prototype.afterGuiAttached = function () {
    this.eInput.focus();
};

// returns the new value after editing
NumericCellEditor.prototype.isCancelBeforeStart = function () {
    return this.cancelBeforeStart;
};

// Gets called once when editing is finished (eg if enter is pressed).
// If you return true, then the result of the edit will be ignored.
NumericCellEditor.prototype.isCancelAfterEnd = function () {
    let value = this.getValue();
    return false;
};

// returns the new value after editing
NumericCellEditor.prototype.getValue = function () {
    return parseInt(this.eInput.value);
};

// any cleanup we need to be done here
NumericCellEditor.prototype.destroy = function () {
    // but this example is simple, no cleanup, we could  even leave this method out as it's optional
};

// if true, then this editor will appear in a popup
NumericCellEditor.prototype.isPopup = function () {
    // and we could leave this method out also, false is the default
    return false;
};

function isCharNumeric(charStr) {
    return !!/\d/.test(charStr);
}

function isKeyPressedNumeric(event) {
    let charCode = getCharCodeFromEvent(event);
    let charStr = String.fromCharCode(charCode);
    return isCharNumeric(charStr);
}

function getCharCodeFromEvent(event) {
    event = event || window.event;
    return (typeof event.which == "undefined") ? event.keyCode : event.which;
}

export function NumericCellEditor() {}


// gets called once before the renderer is used
DecimalCellEditor.prototype.init = function (params) {
    // create the cell
    this.eInput = document.createElement('input');
    params.eGridCell.style.height = '25px'; // make cell stay the same size

    if (isCharNumeric(params.charPress)) {
        this.eInput.value = params.charPress;
    } else {
        if (params.value !== undefined && params.value !== null) {
            this.eInput.value = currency(params.value).value;
        }
    }

    let that = this;
    this.eInput.addEventListener('keypress', function (event) {
        let index = that.eInput.value.indexOf(".");
        if (!isKeyPressedNumericOrDecimal(event) || (index > -1 && (event.key === '.' || that.eInput.value.substring(index).length === 3))) {
            that.eInput.focus();
            if (event.preventDefault) event.preventDefault();
        } else if (that.isKeyPressedNavigation(event)){
            event.stopPropagation();
        }
    });

    // only start edit if key pressed is a number, not a letter
    this.cancelBeforeStart = params.charPress && ('1234567890'.indexOf(params.charPress) < 0);
};

DecimalCellEditor.prototype.isKeyPressedNavigation = function (event){
    return event.keyCode===39
        || event.keyCode===37;
};


// gets called once when grid ready to insert the element
DecimalCellEditor.prototype.getGui = function () {
    return this.eInput;
};

// focus and select can be done after the gui is attached
DecimalCellEditor.prototype.afterGuiAttached = function () {
    this.eInput.focus();
    this.eInput.select(); // highlight existing value
};

// returns the new value after editing
DecimalCellEditor.prototype.isCancelBeforeStart = function () {
    return this.cancelBeforeStart;
};

// Gets called once when editing is finished (eg if enter is pressed).
// If you return true, then the result of the edit will be ignored.
DecimalCellEditor.prototype.isCancelAfterEnd = function () {
    let value = this.getValue();
    return false;
};

// returns the new value after editing
DecimalCellEditor.prototype.getValue = function () {
    return currency(this.eInput.value).value;
};

// any cleanup we need to be done here
DecimalCellEditor.prototype.destroy = function () {
    // but this example is simple, no cleanup, we could  even leave this method out as it's optional
};

// if true, then this editor will appear in a popup
DecimalCellEditor.prototype.isPopup = function () {
    // and we could leave this method out also, false is the default
    return false;
};

function isCharDecimal(charStr) {
    return '.'.indexOf(charStr) === 0;
}

function isKeyPressedNumericOrDecimal(event) {
    let charCode = getCharCodeFromEvent(event);
    let charStr = String.fromCharCode(charCode);
    return isCharNumeric(charStr) || isCharDecimal(charStr);
}

export function DecimalCellEditor() {}