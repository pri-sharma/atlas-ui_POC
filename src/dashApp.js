import React, {Fragment} from 'react';
import {Provider} from 'react-redux';
import {history, store} from './redux/store';
import PublicRoutes from './router';
import {LocaleProvider} from 'antd';
import {IntlProvider} from 'react-intl';
import AppLocale from './languageProvider';
import config, {getCurrentLanguage} from './containers/LanguageSwitcher/config';

const currentAppLocale = AppLocale[getCurrentLanguage(config.defaultLanguage || 'english').locale];

const DashApp = (props) => {
    return (
        <Fragment>
            <LocaleProvider locale={currentAppLocale.antd}>
                <IntlProvider
                    locale={currentAppLocale.locale}
                    messages={currentAppLocale.messages}
                >
                    <Provider store={store}>
                        <PublicRoutes history={history}/>
                    </Provider>
                </IntlProvider>
            </LocaleProvider>
        </Fragment>
    )
};
// Boot()
//     .then(() => DashApp())
//     .catch(error => console.error(error));


export default DashApp;
export {AppLocale};
