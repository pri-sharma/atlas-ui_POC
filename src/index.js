import React from 'react';
import ReactDOM from 'react-dom';
import DashApp from './dashApp';
import * as serviceWorker from './serviceWorker';
import 'antd/dist/antd.css';
import {store} from './redux/store';
import {firebase, googleAuthProvider} from './firebase/firebase';
import {login, logout} from './redux/auth/actions';
import {saveFirebaseToken} from './redux/userSettings/actions';

import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-balham.css';
import 'ag-grid-community/dist/styles/ag-theme-bootstrap.css';
import 'ag-grid-community/dist/styles/ag-theme-material.css';
import {LicenseManager} from 'ag-grid-enterprise';
import '@material-ui/icons';
import * as plannableCustomerActions from "./redux/plannableCustomers/actions";

LicenseManager.setLicenseKey(process.env.REACT_APP_AGGRID_KEY);

let isRendered = false;
const renderApp = () => {
    if (!isRendered) {
        ReactDOM.render(<DashApp/>, document.getElementById('root'));
        isRendered = true;
    }
};

/**
 * Generate token for auth user (force refresh) and store in localstorage/redux
 * @param user
 */
const generateAndStoreToken = user => {
    user.getIdToken(true).then((idToken) => {
        localStorage.setItem('idToken', idToken);
        store.dispatch(login(user.uid, user.email, user.photoURL, idToken));
        store.dispatch(saveFirebaseToken({"token": localStorage.getItem('fcmToken')}));
        store.dispatch(plannableCustomerActions.getPlannableCustomersAction(user.email));
    });
};

let interval;
firebase.auth().onAuthStateChanged(user => {
    if (user) {
        generateAndStoreToken(user);
        clearInterval(interval); // clear existing interval if exists
        interval = setInterval(() => generateAndStoreToken(user), 1000 * 60 * 45); // set interval to refresh token before it expires
    } else {
        firebase.auth().signInWithRedirect(googleAuthProvider).then();
    }
    renderApp();
});

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.register();
