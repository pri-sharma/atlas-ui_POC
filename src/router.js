import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { ConnectedRouter } from 'react-router-redux';
import { connect } from 'react-redux';

import App from './containers/App/App';
import asyncComponent from './helpers/AsyncFunc';
import {restrictedRoutes} from './containers/App/AppRouter';

const RestrictedRoute = ({ component: Component, isLoggedIn, ...rest }) => {
    return (
        <Route
            {...rest}
            render={props => {
                return isLoggedIn ? (
                    <Component {...props} isLoggedIn={isLoggedIn}/>
                ) : (
                    <Redirect
                        to={{
                            pathname: '/signin',
                            state: {from: props.location}
                        }}
                    />
                )}
            }/>);
};
const appRoutes = (isLoggedIn) => {
    return restrictedRoutes.map(route => {
        return (<RestrictedRoute
                exact
                key={route.path}
                path={`${route.path}`}
                component={App}
                isLoggedIn={isLoggedIn}
            />
        )
    })
};
const PublicRoutes = ({ history, isLoggedIn }) => {
  return (
    <ConnectedRouter history={history}>
        <div>
            {appRoutes(isLoggedIn)}
            <Route
                exact
                path={'/signin'}
                component={asyncComponent(() => import('./containers/Page/signin'))}
            />
        </div>
    </ConnectedRouter>
  );
};

export default connect(state => ({
  isLoggedIn: localStorage.getItem('idToken') !== null
}))(PublicRoutes);
