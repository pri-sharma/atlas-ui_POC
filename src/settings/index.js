export default {
  apiUrl: 'http://yoursite.com/api/',
};

const siteConfig = {
  siteName: 'Atlas',
  siteIcon: 'ion-flash',
  footerText: 'Atlas Â©2019',
};
const themeConfig = {
  topbar: 'themedefault',
  sidebar: 'themedefault',
  layout: 'themedefault',
  theme: 'themedefault',
  material: 'materialtheme',
  light: 'lightTheme',
  normal: 'normalTheme',
  dark: 'darkTheme',
};
const language = 'english';

const jwtConfig = {
  fetchUrl: '/api/',
  secretKey: 'secretKey',
};

export { siteConfig, language, themeConfig, jwtConfig };
