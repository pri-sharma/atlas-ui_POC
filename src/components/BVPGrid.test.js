import React from 'react';
import { shallow } from '../../src/setupTests';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {shallowToJson} from 'enzyme-to-json';
import BVPGrid from './BVPGrid';

describe('Grid Initialization', () => {
    it('BVP ', () => {
        const initialState = {
            planningConfig: {
                BVP: {}
            },
            Bvp: {
                baselineVolumePlans: []
            },
            App: {collapsed: false},
            PlannableCustomers: {customerStartDay: "Monday"}
        };

        const middlewares = [thunk];
        const mockStore = configureMockStore(middlewares);
        const store = mockStore(initialState);
        const component = shallow(<BVPGrid store={store}/>);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});