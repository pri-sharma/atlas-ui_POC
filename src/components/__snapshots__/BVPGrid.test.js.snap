// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Initialization BVP  1`] = `
<BVPGrid
  baselineVolumePlans={Array []}
  collapsed={false}
  customerStartDay="Monday"
  getBvpsAction={[Function]}
  saveBvpsAction={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
